src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs

```diff
@@ -471,41 +472,41 @@ public void RegisterOperationAction(DiagnosticAnalyzer analyzer, Action<Operatio
 
     internal abstract class HostAnalysisScope
     {
-        private readonly ConcurrentDictionary<DiagnosticAnalyzer, AnalyzerActions> _analyzerActions = new ConcurrentDictionary<DiagnosticAnalyzer, AnalyzerActions>();
+        private readonly ConcurrentDictionary<DiagnosticAnalyzer, StrongBox<AnalyzerActions>> _analyzerActions = new ConcurrentDictionary<DiagnosticAnalyzer, StrongBox<AnalyzerActions>>();
 
-        public virtual AnalyzerActions GetAnalyzerActions(DiagnosticAnalyzer analyzer)
+        public virtual AnalyzerActions? GetAnalyzerActions(DiagnosticAnalyzer analyzer)
         {
-            return this.GetOrCreateAnalyzerActions(analyzer);
+            return this.GetOrCreateAnalyzerActions(analyzer)?.Value;
```

Comment:
>?.Value [](start = 60, length = 7)

Why not just return directly?

Reply:
Ah, it's a `StrongBox`, not a nullable type.

---
In reply to: [362576725](https://github.com/dotnet/roslyn/pull/40691#discussion_r362576725) [](ancestors = 362576725)